src/roswire/proxy/catkin.py

```diff
@@ -135,13 +171,33 @@ class CatkinMake(CatkinInterface):
     """Provides an interface to a catkin workspace created via catkin_make."""
     directory: str
     _shell: dockerblade.shell.Shell
+    _files: dockerblade.files.FileSystem
 
     def clean(self,
               packages: Optional[List[str]] = None,
               orphans: bool = False,
               context: Optional[str] = None
               ) -> None:
-        raise NotImplementedError
+        shell = self._shell
+        command = ['catkin_make', 'clean']
+        if orphans:
+            raise NotImplementedError
+        if packages:
+            command += ['--pkg']
+            command += [shlex.quote(p) for p in packages]
+        if not context:
+            context = self.directory
+
+        command_str = ' '.join(command)
+        logger.debug(f"cleaning via: {command_str}")
+        result = shell.run(command_str, cwd=context, text=True)
+        duration_mins = result.duration / 60
+        logger.debug("clean completed after %.2f minutes [retcode: %d]:\n%s",
```

Comment:
`%s`, `%.2f`, and so on won't work by default in loguru. We should use f-strings instead:
`clean completed after {durations_mins:.2f} minutes ...`. (Note that you can break Python strings over multiple lines.)